src/parthenon_manager.cpp

```diff
@@ -347,6 +347,10 @@ void ParthenonManager::RestartPackages(Mesh &rm, RestartReader &resfile) {
         // check if the sparse variable is allocated on this block
         if (sparse_info.IsAllocated(pmb->gid, sparse_idxs.at(label))) {
           pmb->AllocateSparse(label);
+          auto dealloc_count = sparse_info.DeallocCount(pmb->gid, sparse_idxs.at(label));
+          // Warning: For this to work, it is required that the controlling variable is
+          // stored in the restart files.
+          pmb->meshblock_data.Get()->GetVarPtr(label)->dealloc_count = dealloc_count;
```

Comment:
Can this reasonable be triggered in practice? And if so, does it make sense to throw here?

Reply:
Actually, I think that I added this warning is erroneously. I think the sparse information for all fields (not just those that are written to file) is output here. I will remove it.